presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoPreparedStatement.java

```diff
@@ -455,7 +458,26 @@ public void setArray(int parameterIndex, Array x)
     public ResultSetMetaData getMetaData()
             throws SQLException
     {
-        throw new SQLFeatureNotSupportedException("getMetaData");
+        try (PrestoStatement statement = (PrestoStatement) getConnection().createStatement()) {
+            ResultSet rs = statement.executeQuery("DESCRIBE OUTPUT " + statementName);
```

Comment:
Add `ResultSet` to the try-with-resources statement so that it is closed properly:
```java
try (Statement statement = getConnection().createStatement();
        ResultSet rs = statement.executeQuery("DESCRIBE OUTPUT " + statementName)) {